### PR TITLE
Fix START TRANSACTION

### PR DESCRIPTION
**Pull request by 1st1**
Please see the individual commits.

### CODE REVIEW

**Review by elprans — APPROVED** [review]

File: edb/server/pgcon/pgcon.pyx [hunk 1]
```diff
@@ -1068,6 +1076,9 @@ cdef class PGConnection:
         try:
             if state is not None:
                 await self._parse_apply_state_resp(state)
+                if state_sync:
+                    await self.wait_for_sync()
+                    self.waiting_for_sync = True
```
Comment (fantix) [inline edb/server/pgcon/pgcon.pyx:1081]: Maybe add a comment/GOTCHA here, specifying the reason why the order of these two lines are not wrong? I was confused in the first place without reading the context.

**Review by fantix — APPROVED** [review]
lgtm!

File: edb/server/pgcon/pgcon.pyx (hunk 1 above)
Comment (1st1) [reply edb/server/pgcon/pgcon.pyx]: Sure, done

**Comment by 1st1** [comment]
@msullivan @elprans please cherry pick to 1.x

**Pull request merged**